src/Logging/PsrLogger.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Logging;
+
+use Psr\Log\LoggerInterface;
+
+class PsrLogger implements LoggerInterface
```

Comment:
When we talked about this before, I think my main reason for preferring them to be separate is that the PSR Logger will need the `$resource` to be specified ahead of time, whereas the regular Logger accepts it as an argument on the `entry()` method.

To fix this problem while keeping everything in the same class, you'd either need a setter `setResource()`, or have an optional constructor argument that would be used in PSR-3-contexts, but not provided (or simply disregarded) in more complex logging operations that don't fix into PSR-3.

````php
// Which gets used? It's a bit ambiguous.
$logger = new Logger('logName', ['resourceThing' => 'foo']);
$logger->entry('message', ['resourceThing' => 'theRealResource']);
````